qemu/tests/ansible_test.py

```diff
@@ -58,27 +57,22 @@ def run(test, params, env):
                   "test_harness_log_dir": test_harness_log_dir}
     extra_vars.update(json.loads(ansible_extra_vars))
 
-    ansible_cmd_options = ["ansible-playbook",
-                           "-u {}".format(guest_user),
-                           "-i {},".format(",".join(guest_ip_list)),
-                           "-e '{}'".format(json.dumps(extra_vars)),
-                           ansible_addl_opts,
-                           toplevel_playbook]
-    ansible_cmd = r" ".join(ansible_cmd_options)
-
     error_context.context("Execute the ansible playbook.", logging.info)
-    env_vars = ({"ANSIBLE_STDOUT_CALLBACK": ansible_callback_plugin}
-                if ansible_callback_plugin else None)
-    logging.info("Command of ansible playbook: '%s'", ansible_cmd)
-    play_s, play_o = process.getstatusoutput(ansible_cmd,
-                                             timeout=playbook_timeout,
-                                             shell=False, env=env_vars)
-    ansible_log = "ansible_playbook.log"
-    with open(os.path.join(test_harness_log_dir, ansible_log), "w") as log_file:
-        log_file.write(play_o)
-        log_file.flush()
+    playbook_executor = PlaybookExecutor(
+        inventory="{},".format(",".join(guest_ip_list)),
+        site_yml=toplevel_playbook,
+        remote_user=guest_user,
+        extra_vars=json.dumps(extra_vars),
+        callback_plugin=ansible_callback_plugin,
+        addl_opts=ansible_addl_opts
+    )
+
+    playbook_executor.wait_for_completed(playbook_timeout, step_time)
```

Comment:
Need handle playbook timeout error. Now just logs the error message.

Reply:
After the ansible-playbook terminated, only logs the error message, the full output will store into ansible_playbook.log 
```
playbook_executor.store_playbook_log(test_harness_log_dir, ansible_log)
```

Reply:
Updated, please help review again, thanks.